charts/newrelic-infrastructure/templates/controlplane/daemonset.yaml

```diff
@@ -97,6 +103,18 @@ spec:
           {{- with .Values.controlPlane.extraEnvFrom }}
           envFrom: {{ toYaml . | nindent 12 }}
           {{- end }}
+
+          {{- if not (include "nriKubernetes.controlPlane.hostNetwork" .) }}
+          ports:
+            - containerPort: 8999
```

Comment:
Would it worth it to make the health port configurable?

Reply:
I've started implementing this, but I'm finding UX concerns that are pretty hard to solve. The main one is that, right now, it's easy to expose the `containerPort` and to disable everything if `hostNetwork` is on. If we expose the config, that logic gets more complicated, especially if we want to override this setting depending on `hostNetwork`.

Reply:
I'm not a big fan of the hardcoded probe as it is now either... Any ideas, @kang-makes ?

Reply:
It should listen to `localhost` and it is disabled when `hostNetwork` is enabled so I have no big issues with hardcoding it.

As it could be set in the configuration file we could test in the `.Values.common.config` has a `healthAddress` and set it.

Reply:
Is there any other already configurable port which may collide if it is set to `8999`? If we finally keep it hardcoded, that is the only cause for concern I can think of...

Reply:
There is no configurable port in our helm charts. We could configure LOTS of things so we timeboxed and we run out of time.

It is prepared for the port to be variable but right now it is hardcoded in helpers:
https://github.com/newrelic/nri-kubernetes/blob/main/charts/newrelic-infrastructure/templates/controlplane/_agent-config_helper.tpl#L7
https://github.com/newrelic/nri-kubernetes/blob/main/charts/newrelic-infrastructure/templates/ksm/_agent-config_helper.tpl#L7
https://github.com/newrelic/nri-kubernetes/blob/main/charts/newrelic-infrastructure/templates/kubelet/_agent-config_helper.tpl#L6

Reply:
I wanted to have preview but Github seems to be drunk (again)

Ports 8001, 8002, and 8003 are hardcoded right now.